### PR TITLE
WIP: Fix phpunit notices

### PR DESCRIPTION
**Pull request by josbeir**
This PR fixes the various PHPunit notices that started to pop up in 12.5 when using `createMock` with no expectations.

I focussed on quick wins where `createMock` could relatively easily be swapped with `createStub`. The more complex situations are handled with the `AllowMockObjectsWithoutExpectations` attribute.

- [x] Phase 1: Paginator Tests (124 notices)
  - [x] PaginatorTestTrait  (62) 
- [x] Phase 2: ORM Behavior Tests (16 notices)
  - [x] BehaviorTest.php (16) 
- [x] Phase 3: Database Type Tests (226 notices)
  - [x] DateTimeTimezoneTypeTest.php (52) 
  - [x] DateTimeFractionalTypeTest.php (49) 
  - [x] DateTimeTypeTest.php (42) 
  - [x] TimeTypeTest.php (29) 
  - [x] DateTypeTest.php (28) 
  - [x] DriverTest.php (25) 
- [x] Phase 4: ORM Association Tests (160 notices)
  - [x] BelongsToManyTest.php (73)
  - [x] HasManyTest.php (60)
  - [x] AssociationTest.php (27)
- [x] Phase 5: Database Schema Tests (268 notices)
  - [x] SqliteSchemaDialectTest.php (96)
  - [x] MysqlSchemaDialectTest.php (51)
  - [x] PostgresSchemaDialectTest.php (40)
  - [x] SqlserverSchemaDialectTest.php (30)
- [x] Phase 6: HTTP & Remaining Tests (31+ notices)
  - [x] StreamTest.php (12)
  - [x] ResponseEmitterTest.php (10)
  - [x] RateLimitMiddlewareTest.php (9)
- [x] Phase 7: TableTest.php (14 notices)
- [x] Phase 8: Remaining Type & Driver Tests (120 notices)
  - [x] Database Type Tests:
    - [x] IntegerTypeTest.php (9 notices)
    - [x] JsonTypeTest.php (8 notices)
    - [x] FloatTypeTest.php (7 notices)
    - [x] BoolTypeTest.php (7 notices)
    - [x] UuidTypeTest.php (5 notices)
    - [x] StringTypeTest.php (5 notices)
    - [x] BinaryUuidTypeTest.php (5 notices)
    - [x] BinaryTypeTest.php (4 notices)
  - [x] Database Driver/Factory Tests:
    - [x] SqliteTest.php (9 notices)
    - [x] TypeFactoryTest.php (5 notices)
    - [x] SqlserverTest.php (5 notices)
    - [x] PostgresTest.php (3 notices)
  - [x] Console Tests:
    - [x] CommandRunnerTest.php (9 notices)
    - [x] ConsoleOutputTest.php (4 notices)
    - [x] SchemaCacheCommandsTest.php (4 notices)
  - [x] ORM Tests:
    - [x] AssociationCollectionTest.php (7 notices)
    - [x] TableLocatorTest.php (3 notices)
    - [x] TableRegistryTest.php (1 notice)
    - [x] Other ORM tests (6 notices total)
  - [x] Other Tests (9 notices)

### CODE REVIEW

File: tests/TestCase/Datasource/Paging/PaginatorTestTrait.php [hunk 1]
```diff
@@ -85,20 +79,18 @@ protected function tearDown(): void
     /**
      * Test that non-numeric values are rejected for page, and limit
      */
+    #[AllowMockObjectsWithoutExpectations]
```
Comment (ADmad) [inline tests/TestCase/Datasource/Paging/PaginatorTestTrait.php:82]: The attribute shouldn't be needed since the code was updated to use `createStub()` isn't it?

Comment (josbeir) [reply tests/TestCase/Datasource/Paging/PaginatorTestTrait.php:82]: This is an edge case since it still needs a mock to allow chaining

Comment (josbeir) [reply tests/TestCase/Datasource/Paging/PaginatorTestTrait.php:82]: added reasoning why its used per file in the PR message

**Comment by ADmad** [comment]
> 1) Cake\Test\TestCase\Cache\Engine\RedisClusterEngineTest::testConnectRedisClusterFailureLogsError
No expectations were configured for the mock object for RedisCluster. Consider refactoring your test code to use a test stub instead. The #[AllowMockObjectsWithoutExpectations] attribute can be used to opt out of this check.

> /home/runner/work/cakephp/cakephp/tests/TestCase/Cache/Engine/RedisClusterEngineTest.php:191

Seems there's one remaining

**Comment by josbeir** [comment]
> Seems there's one remaining

Fixed, didn't notice it because it was in a skipped test on my local machine :-)

**Review by ADmad — APPROVED** [review]
